public/app/plugins/datasource/elasticsearch/components/QueryEditor/index.test.tsx

```diff
@@ -45,5 +45,41 @@ describe('QueryEditor', () => {
       expect(onChange).toHaveBeenCalledTimes(1);
       expect(onChange.mock.calls[0][0].alias).toBe(newAlias);
     });
+
+    it('Should be disabled if last bucket aggregation is not Date Histogram', () => {
+      const query: ElasticsearchQuery = {
+        refId: 'A',
+        query: '',
+        metrics: [
+          {
+            id: '1',
+            type: 'avg',
+          },
+        ],
+        bucketAggs: [{ id: '2', type: 'terms' }],
+      };
+
+      render(<QueryEditor query={query} datasource={{} as ElasticDatasource} onChange={noop} onRunQuery={noop} />);
+
+      expect(screen.getByLabelText('Alias')).toBeDisabled();
+    });
+
+    it('Should be enabled if last bucket aggregation is Date Histogram', () => {
+      const query: ElasticsearchQuery = {
+        refId: 'A',
+        query: '',
+        metrics: [
+          {
+            id: '1',
+            type: 'avg',
+          },
+        ],
+        bucketAggs: [{ id: '2', type: 'date_histogram' }],
+      };
+
+      render(<QueryEditor query={query} datasource={{} as ElasticDatasource} onChange={noop} onRunQuery={noop} />);
+
+      expect(screen.getByLabelText('Alias')).not.toBeDisabled();
```

Comment:
just a  (personal) nit:

```
      expect(screen.getByLabelText('Alias')).not.toBeDisabled();
```

or

```
      expect(screen.getByLabelText('Alias')).toBeEnabled();
```
?

Reply:
I would probably go with toBeEnabled but this is just personal preference.

Reply:
will update! thanks! ❤️